### PR TITLE
Add profile filtering when adding Generic Entity IDs in Oncoprint

### PR DESCRIPTION
**Pull request by oplantalech**
Fix cBioPortal/cbioportal#10078

### CODE REVIEW

File: src/pages/resultsView/oncoprint/TracksMenu.tsx [hunk 1]
```diff
@@ -60,9 +67,34 @@ export const COUNT_PADDING_WIDTH = 17;
...
+        string,
+        string
+    >({}, { deep: true });
+    @observable XvsYSelection: {
+        x?: { value: string; label: string };
+        y?: { value: string; label: string };
+    } = {
+        x: undefined,
+        y: undefined,
+    };
```
Comment (dippindots) [inline src/pages/resultsView/oncoprint/TracksMenu.tsx:82]: Is this one necessary? Maybe we can delete it if it is not used by others.

File: src/pages/resultsView/oncoprint/TracksMenu.tsx [hunk 2]
```diff
@@ -394,12 +426,23 @@ export default class TracksMenu extends React.Component<IAddTrackProps, {}> {
...
+        this.selectedGenericAssayProfileIdByType.set(
+            genericAssayType,
+            profileId
+        );
+    }
+
     @computed
     private get genericAssayTabs() {
         let tabs = [];
         if (this.isGenericAssayDataComplete && this.showGenericAssayTabs) {
```
Comment (dippindots) [inline src/pages/resultsView/oncoprint/TracksMenu.tsx:443]: This is my fault, it looks like I forgot to add necessary checks in `isGenericAssayDataComplete` function.
Could you add `this.props.store.genericAssayEntitiesGroupByMolecularProfileId.isComplete` and `this.selectedGenericAssayProfileIdByType.size > 0` into `isGenericAssayDataComplete` function?

File: src/pages/resultsView/oncoprint/TracksMenu.tsx [hunk 3]
```diff
@@ -60,9 +67,34 @@ export const COUNT_PADDING_WIDTH = 17;
...
+        y?: { value: string; label: string };
+    } = {
+        x: undefined,
+        y: undefined,
+    };
+
     constructor(props: IAddTrackProps) {
         super(props);
         makeObservable(this);
+        // Using the first profile as default profile option to initialize genericAssayProfileOptionsByType
```
Comment (dippindots) [inline src/pages/resultsView/oncoprint/TracksMenu.tsx:87]: Maybe change text to this?
```suggestion
        // Using the first profile as default profile option to initialize selectedGenericAssayProfileIdByType
```

**Review by dippindots — COMMENTED** [review]
@oplantalech Thanks again for fixing this issue, it looks good to me, just had some minor comments.

**Comment by oplantalech** [comment]
@dippindots Thank you for your review! I've implemented all your suggestions except one small thing (see comment). When it's all ready let me know and I'll squash all the commits before merging

File: src/pages/resultsView/oncoprint/TracksMenu.tsx [hunk 4]
```diff
@@ -60,9 +67,27 @@ export const COUNT_PADDING_WIDTH = 17;
...
+    private selectedGenericAssayProfileIdByType = observable.map<
+        string,
+        string
+    >({}, { deep: true });
+
     constructor(props: IAddTrackProps) {
         super(props);
         makeObservable(this);
+        // Using the first profile as default profile option to initialize selectedGenericAssayProfileIdByType
+        this.genericAssayEntityOptionsReaction = when(
```
Comment (alisman) [inline src/pages/resultsView/oncoprint/TracksMenu.tsx:81]: not sure why this reaction is necessary

**Review by alisman — CHANGES_REQUESTED** [review]
@oplantalech see my comment regarding the reaction

**Review by alisman — CHANGES_REQUESTED** [review]
@oplantalech see my comment regarding the reaction

**Review by alisman — CHANGES_REQUESTED** [review]
@oplantalech see my comment regarding the reaction

File: src/pages/resultsView/oncoprint/TracksMenu.tsx (hunk 4 above)
Comment (oplantalech) [reply src/pages/resultsView/oncoprint/TracksMenu.tsx]: @alisman Thanks for this comment. I've simplified the code deleting the reaction

File: src/pages/resultsView/oncoprint/TracksMenu.tsx [hunk 5]
```diff
@@ -60,6 +67,13 @@ export const COUNT_PADDING_WIDTH = 17;
 export default class TracksMenu extends React.Component<IAddTrackProps, {}> {
     @observable tabId: Tab | string = Tab.CLINICAL;
 
+    private genericAssayEntityOptionsReaction: IReactionDisposer;
```
Comment (alisman) [inline src/pages/resultsView/oncoprint/TracksMenu.tsx:70]: delete

File: src/pages/resultsView/oncoprint/TracksMenu.tsx [hunk 6]
```diff
@@ -60,6 +67,13 @@ export const COUNT_PADDING_WIDTH = 17;
...
+
+    private selectedGenericAssayProfileIdByType = observable.map<
```
Comment (alisman) [inline src/pages/resultsView/oncoprint/TracksMenu.tsx:72]: is it possible to have multiple profiles selected?

Comment (oplantalech) [reply src/pages/resultsView/oncoprint/TracksMenu.tsx:63]: You always have one profile selected per Generic Assay Type: 
Screenshot from 2023-05-10 12-15-53

So in this example there are three Generic Assay Types (Generic Assay Patient Test, Treatment Response and Mutational Signature) and both have one profile selected. You cannot have more than one profile selected per Generic Assay Type.

**Review by alisman — APPROVED** [review]

**Comment by oplantalech** [comment]
@alisman Can we merge?

**Comment by dippindots** [comment]
@oplantalech Merged! Really appreciate you fixing this issue!

**Pull request merged**